lib/surface/live_component.ex

```diff
@@ -65,20 +65,16 @@ defmodule Surface.LiveComponent do
       @behaviour unquote(__MODULE__)
       require Surface.ContentHandler
       @before_compile Surface.ContentHandler
-      Module.put_attribute(__MODULE__, :__is_stateful__, true)
+
+      @doc """
+      The id of the live component (required by LiveView).
+      """
+      property id, :string, required: true
```

Comment:
Should it be required? As far as I can remember LV only requires an id when using `handle_event`. Has that changed?

Reply:
You're correct that LV only requires it when using `handle_event`; however, we're treating `LiveComponent` as if it's stateful and using the `@myself` assign. That assign is only created if the component has an id (and is therefore stateful). I pushed up a test showing the issue

Reply:
I see. So ideally we could always define the property but make it required only when the component has either a `handle_event` or defines a property of type `:event`. Is there any other case?

Reply:
Those are the only two cases I've run into - if we wanted to keep it optional when it wasn't strictly required by LV, that should be sufficient.

My rational on keeping it always required would be that we call `LiveComponent` stateful and `Component` stateless in our docs, and the thing that makes it stateful in LV is the `id` prop. I don't have a strong preference though 🙂

Reply:
Having the `id` required when we know it's required is nice because we add static validation while LV will only complain at runtime. However, if LV allows LiveComponents without `id`, I think we should allow that too. I believe there must be cases when this is useful. I'm just not sure when. 🙂

Reply:
that makes sense - i think my assumption has been that a `Surface.LiveComponent` without an id is functionally equivalent to a `Surface.Component`

Reply:
It is equivalent today because of a limitation of LV which does not work properly when you have a `LiveComponent` inside a regular `Component` (I believe is the same issue that doesn't allow using `LiveComponent` inside `content_tag`). My initial goal was to make `Component` available to non-LV projects. Actually the first version didn't depend on LV at all. Unfortunately, I had to implement `Component` as a `LiveComponent` later just to "fix" this issue. 

I'd still love to have a real stateless component that could be used in non-LV projects. It would be very useful.

Reply:
That makes sense - I'll see how complicated it'd be to switch it to required only if handle_event is exported or an :event property is used

Reply:
> switch it to required only if handle_event is exported or an :event property is used

Regarding checking the `:event` property, that might be tricky. The `id` would be required only if the live component passes an event to a child component as a string, which is basically the same as passing `{{ ..., target: @myself }}`.

Maybe it's better to leave that one out for now. We can try it later for `v.0.2.0` or higher.

Reply:
That sounds reasonable, it's easier to go from required to optional than optional to required. In that case, this PR may be ready to merge